x/crosschain/keeper/keeper_cross_chain_tx_prove_outbound_tx.go

```diff
@@ -0,0 +1,195 @@
+//
+// If the observation is successful, the difference between zeta burned
+// and minted is minted by the bank module and deposited into the module
+// account.
+//
+// If the observation is unsuccessful, the logic depends on the previous
+// status.
+//
+// If the previous status was `PendingOutbound`, a new revert transaction is
+// created. To cover the revert transaction fee, the required amount of tokens
+// submitted with the CCTX are swapped using a Uniswap V2 contract instance on
+// ZetaChain for the ZRC20 of the gas token of the receiver chain. The ZRC20
+// tokens are then
+// burned. The nonce is updated. If everything is successful, the CCTX status is
+// changed to `PendingRevert`.
+//
+// If the previous status was `PendingRevert`, the CCTX is aborted.
+//
+// ```mermaid
+// stateDiagram-v2
+//
+//	state observation <<choice>>
+//	state success_old_status <<choice>>
+//	state fail_old_status <<choice>>
+//	PendingOutbound --> observation: Finalize outbound
+//	observation --> success_old_status: Observation succeeded
+//	success_old_status --> Reverted: Old status is PendingRevert
+//	success_old_status --> OutboundMined: Old status is PendingOutbound
+//	observation --> fail_old_status: Observation failed
+//	fail_old_status --> PendingRevert: Old status is PendingOutbound
+//	fail_old_status --> Aborted: Old status is PendingRevert
+//	PendingOutbound --> Aborted: Finalize outbound error
+//
+// ```
+//
+// Only observer validators are authorized to broadcast this message.
+func (k msgServer) ProveOutboundTx(goCtx context.Context, msg *types.MsgProveOutboundTx) (*types.MsgProveOutboundTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// verify proof
+	blockHash := eth.HexToHash(msg.BlockHash)
+	res, found := k.zetaObserverKeeper.GetBlockHeader(ctx, blockHash.Bytes())
+	if !found {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "Block header not found: %s", msg.BlockHash)
+	}
+	chainID := res.ChainId
+
+	var header ethtypes.Header
+	err := rlp.DecodeBytes(res.Header, &header)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "cannot decode block header: %s", msg.BlockHash)
+	}
+	var txx ethtypes.Transaction
+	var receipt ethtypes.Receipt
+	txxVal, err := msg.TxProof.Verify(header.TxHash, int(msg.TxIndex))
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot verify tx proof: %s", err)
+	}
+	receiptVal, err := msg.ReceiptProof.Verify(header.ReceiptHash, int(msg.TxIndex))
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot verify receipt proof: %s", err)
+	}
+	err = txx.UnmarshalBinary(txxVal)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot unmarshal tx: %s", err)
+	}
+	err = receipt.UnmarshalBinary(receiptVal)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot unmarshal receipt: %s", err)
+	}
```

Comment:
I think it would be more visual to put it in a specific method.
It would also simplify unit testing, we should have extensive unit testing of this functionality since it define the logic of validating a tx.